src/apps/pillarx-app/components/WalletPortfolioButtons/WalletPortfolioButtons.tsx

```
@@ -38,23 +44,157 @@ const WalletPortfolioButtons = () => {
+        },
+      });
+
+      if (!tokenResponse.ok) {
+        // eslint-disable-next-line no-console
+        console.error(
+          'Token response not OK:',
+          tokenResponse.status,
+          tokenResponse.statusText
+        );
+        return;
+      }
+
+      const tokenData = await tokenResponse.json();
+      const { token } = tokenData;
+
+      if (!token) {
+        // eslint-disable-next-line no-console
+        console.error('No token in response:', tokenData);
+        return;
+      }
+
+      // Get user's current IP address
+      let clientIp = '127.0.0.1'; // fallback
+      try {
+        const ipResponse = await fetch('https://api.ipify.org?format=json');
+        if (ipResponse.ok) {
+          const ipData = await ipResponse.json();
+          clientIp = ipData.ip;
+        }
+      } catch (ipError) {
+        // eslint-disable-next-line no-console
+        console.warn('Failed to get IP, using fallback:', ipError);
+      }
+
+      // Call Coinbase API to create onramp session (via proxy to avoid CORS)
+      // This uses Vite proxy in dev and Cloudflare Pages Function in production
+      const response = await fetch(
+        '/api/coinbase/platform/v2/onramp/sessions',
+        {
+          method: 'POST',
+          headers: {
+            Authorization: `Bearer ${token}`,
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify({
+            purchaseCurrency: 'USDC',
+            destinationNetwork: 'base',
+            destinationAddress: accountAddress,
+            paymentCurrency: 'USD',
+            clientIp,
+            redirectUrl: window.location.origin,
+          }),
+        }
+      );
+
+      const data = await response.json();
+
+      if (!response.ok) {
+        const errorMsg = `Coinbase API error (${response.status}): ${data.error || JSON.stringify(data)}`;
+        // eslint-disable-next-line no-console
+        console.error(errorMsg);
```

This looks like another silent error - alert the user that something went wrong here and they should try again